### PR TITLE
Add Tendencies class

### PR DESCRIPTION
**Pull request by sbrus89**
This PR adds a class that provides routines to compute groups of tendencies for thickness and velocity. It is dependent on the completion of `AuxilaryState` so I've put it up as a draft PR for now.

Checklist
* [x] Documentation:
  * [x] Design document has been generated and added to the docs
  * [x] User's Guide has been updated
  * [x] Developer's Guide has been updated
  * [x] Documentation has been [built locally](https://e3sm-project.github.io/Omega/develop/devGuide/BuildDocs.html) and changes look as expected
* [x] Testing
  * [x] A comment in the PR documents testing used to verify the changes including any tests that are added/modified/impacted.
  * [x] CTest unit tests for new features have been added per the approved design. 
  * [x] Unit tests have passed. Please provide a relevant CDash build entry for verification.
before and after.

### CODE REVIEW

File: components/omega/src/ocn/TendencyTerms.cpp [hunk 1]
```diff
@@ -12,9 +12,197 @@
...
+} // end thickness tehndency compute
+
+//------------------------------------------------------------------------------
+// Compute tendencies for normal velocity equation
+// TODO Add AuxilaryState as argument
+void Tendencies::computeVelocityTendencies(
+    OceanState *State ///< [in] State variables
+) {
+
+   OMEGA_SCOPE(o_NormalVelocityTend, NormalVelocityTend);
```
Comment (mwarusz) [inline components/omega/src/ocn/TendencyTerms.cpp:141]: See https://github.com/E3SM-Project/Omega/pull/92#discussion_r1670441698. We need to pick a convention and stick to it.

**Review by mwarusz — CHANGES_REQUESTED** [review]
This is a good start. I will do a more thorough review when it starts using the auxiliary state. Just some small comments for now.

File: components/omega/src/ocn/TendencyTerms.cpp (hunk 1 above)
Comment (sbrus89) [reply components/omega/src/ocn/TendencyTerms.cpp]: Good point, I went to using the `Loc` prefix you used in #112.

File: components/omega/src/ocn/TendencyTerms.h [hunk 2]
```diff
@@ -261,5 +263,79 @@ class VelocityHyperDiffOnEdge {
...
+ public:
+   // Arrays for accumulating tendencies
+   Array2DReal LayerThicknessTend;
+   Array2DReal NormalVelocityTend;
+
+   // Instances of tendency terms
+   ThicknessFluxDivOnCell ThicknessFluxDiv;
+   PotentialVortHAdvOnEdge PotientialVortHAdv;
+   KEGradOnEdge KEGrad;
+   SSHGradOnEdge SHHGrad;
```
[the remark this answers is not in the document]
Comment (sbrus89) [reply components/omega/src/ocn/TendencyTerms.h:278]: Thanks for catching this, @mwarusz.

**Comment by sbrus89** [comment]
**Testing**

This passes the unit tests on Frontier

**Review by mwarusz — APPROVED** [review]
Thanks @sbrus89 for addressing my comments.

Unit tests pass on Chrysalis and PM-CPU/GPU. This PR looks good to me. Further changes needed for time stepping can be applied in #120.

**Review by philipwjones — APPROVED** [review]
Other than my usual objections over the granularity of these calculations, this looks fine and passes testing.

**Review by brian-oneill — APPROVED** [review]
Looks great, tests pass successfully. Organizationally, I was thinking that the Tendency class and the individual terms would be in separate code files. If we keep them in the same files like this, maybe makes sense to rename TendencyTerms.cpp/.h to Tendencies.cpp/.h. Regardless, I'm approving to go forward with merge.

**Comment by sbrus89** [comment]
@brian-oneill, I like the idea of putting them in separate files. @mwarusz had the same opinion. We can do that in a follow-on PR.

**Pull request merged**